verify_rpms/rpm_verifier.py

```diff
@@ -44,85 +47,85 @@ def get_rpmdb(container_image: str, target_dir: Path, runner: Callable = run) ->
+            "%{NAME}-%{VERSION}-%{RELEASE} %{SIGPGP:pgpsig}\n",
             "--dbpath",
             str(rpmdb),
         ],
         capture_output=True,
         text=True,
         check=True,
     ).stdout.splitlines()
-    return [
+    return rpm_strs
+
+
+def get_unsigned_rpms(rpms: list[str]) -> list[str]:
+    """
+    Get all unsigned RPMs from the list of RPMs
+    Filters out the `gpg-pubkey` and return the unsigned RPMs
+    :param rpms: list of RPMs
+    :return: list of unsigned RPMs
+    """
+    unsigned_rpms: list[str] = [
         rpm.split()[0]
-        for rpm in rpm_strs
+        for rpm in rpms
         if "Key ID" not in rpm and not rpm.startswith("gpg-pubkey")
     ]
 
+    return unsigned_rpms
 
-def generate_output(processed_images: Iterable[ProcessedImage]) -> tuple[bool, str]:
+
+def generate_output(processed_image: ProcessedImage) -> tuple[bool, str]:
     """
     Generate the output that should be printed to the user based on the processed images
     results.
-
-    :param processed_images: each element contains a name and a list of unsigned images
-
+    :param processed_image: ProcessedImage that contains all the rpms information
     :returns: The status and summary of the image processing.
     """
-    failures = True
-    with_unsigned_rpms = [img for img in processed_images if img.unsigned_rpms]
-    with_error = [img for img in processed_images if img.error]
-    if not with_unsigned_rpms and not with_error:
-        output = "No unsigned RPMs found."
-        failures = False
-    else:
-        output = "\n".join([str(img) for img in with_unsigned_rpms])
-        output = f"Found unsigned RPMs:\n{output}" if with_unsigned_rpms else ""
-        error_str = "\n".join([f"{img.image}: {img.error}" for img in with_error])
-        output = f"{output}\nEncountered errors:\n{error_str}" if error_str else output
-    return failures, output.lstrip()
-
+    failure = True
 
-def parse_image_input(image_input: str) -> list[str]:
-    """
-    Input is either an image reference or snapshot data in json format.
-    Try parsing as json and extract the images.
-    If failing to parse as json, assume it's an image reference as one-element list.
-    """
-    try:
-        snapshot = loads(s=image_input)
-    except JSONDecodeError:
-        return [image_input]
-    return [component["containerImage"] for component in snapshot["components"]]
+    if not processed_image.unsigned_rpms and not processed_image.error:
+        output = f"No unsigned RPMs in {processed_image.image}"
+        failure = False
+    else:
+        output = (
+            f"Found unsigned RPMs in {processed_image.image}:\n{processed_image.unsigned_rpms}"
+            if processed_image.unsigned_rpms
+            else ""
+        )
+        output = (
+            f"Encountered errors in {processed_image.image}:\n{processed_image.error}"
+            if processed_image.error
+            else output
+        )
```

Comment:
With what you mention about the previous comment, this could have been something like:
```suggestion
        output = (
            f"Found unsigned RPMs in {processed_image.image}:\n{processed_image.unsigned_rpms}"
            if processed_image.unsigned_rpms
            else f"Encountered errors in {processed_image.image}:\n{processed_image.error}"
        )
```

Reply:
Done